src/Tracker/index.tsx

```diff
@@ -116,16 +117,33 @@ const appendToTracking = <P extends WithTrackingProps>(
     )
   }
 
+type AnalyticsDynamicProperties = Record<string, unknown>
+
 const trackComponent = <P extends WithTrackingProps>(
   WrappedComponent: ComponentType<P>,
   screenName?: string
 ): ComponentType<P> =>
   class TrackedComponent extends Component<P> {
+    callback?: () => AnalyticsDynamicProperties = undefined
     componentDidMount() {
-      this.props.trackScreen(screenName)
+      const properties =
+        typeof this.callback === 'function' ? this.callback() : undefined
```

Comment:
what if properties were a prop on the component itself, similar to the trackScreen ? this way we could 
```suggestion
      const properties = this.props.trackPropertiesBeforeMount ? this.props.trackPropertiesBeforeMount() : undefined
```
But we would need to update withTrackingProps